src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -199,7 +199,7 @@ protected override Task WriteResponseHeadersAsyncCore(Metadata responseHeaders)
                 }
             }
 
-            return HttpContext.Response.Body.FlushAsync();
+            return HttpContext.Response.StartAsync();
```

Comment:
This isn't right. You want to keep calling FlushAsync

Reply:
Hmm does StartAsync not flush the headers?

Reply:
No, it does not.

Reply:
Is Body.FlushAsync right tho? Everywhere else we're all about BodyPipe.

`BodyPipe.FlushAsync()`? (or whatever its pipe equivalent is)

Reply:
I dunno if we implemented this. @jkotalik if you do a flush without writing any body, do we flush the headers? Maybe we should add something first class on the HttpResponse...

Reply:
At one point I thought StartAsync would actually write the headers not just commit them. I could be mis-remembering though. Regardless the FlushAsync API should still work? cc @halter73

Reply:
Flushing the response pipe should flush headers. StartAsync just commits them.

Reply:
I think I'll leave the implementation as is for now as that should still work. I see some test failures when I try to switch to BodyPipe.FlushAsync including HasStarted but I think it's a test. I'll investigate it later since I need this merged ASAP.